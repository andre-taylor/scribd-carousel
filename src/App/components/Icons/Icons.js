import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './Icons.module.css';

export const IconArrow = ({ direction }) => {
  return <span className={cx(styles.arrow, styles[direction])}></span>;
};

IconArrow.defaultProps = {
  direction: 'arrowLeft',
};

// only support's left and right arrows
IconArrow.propTypes = {
  direction: PropTypes.oneOf(['arrowLeft', 'arrowRight', 'arrowDisabled']),
};
