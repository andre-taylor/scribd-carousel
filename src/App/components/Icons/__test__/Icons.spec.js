import React from 'react';
import renderer from 'react-test-renderer';

import { IconArrow } from '../Icons';

describe('<IconArrow />', () => {
  it('renders correctly with defaultProps', () => {
    const component = renderer.create(<IconArrow />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with direction=arrowLeft', () => {
    const component = renderer.create(<IconArrow direction={'arrowLeft'} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with direction=arrowRight', () => {
    const component = renderer.create(<IconArrow direction={'arrowRight'} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with direction=arrowDisabled', () => {
    const component = renderer.create(
      <IconArrow direction={'arrowDisabled'} />
    );
    expect(component).toMatchSnapshot();
  });
});
