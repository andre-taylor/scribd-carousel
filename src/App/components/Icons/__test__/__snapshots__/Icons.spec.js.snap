// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconArrow /> renders correctly with defaultProps 1`] = `
<span
  className="arrow arrowLeft"
/>
`;

exports[`<IconArrow /> renders correctly with direction=arrowDisabled 1`] = `
<span
  className="arrow arrowDisabled"
/>
`;

exports[`<IconArrow /> renders correctly with direction=arrowLeft 1`] = `
<span
  className="arrow arrowLeft"
/>
`;

exports[`<IconArrow /> renders correctly with direction=arrowRight 1`] = `
<span
  className="arrow arrowRight"
/>
`;
